SPARCbot.py

```diff
@@ -168,8 +184,276 @@ async def iwantto(self, ctx, role: str):
     async def unsure(self, ctx):
         await timed_send(ctx, 'I\'m afraid I can\'t let you do that.')
 
+class Bets(commands.Cog):
 
 
+    def __init__(self,bot):
+        self.bot = bot
+        self.lock = False
+
+    def add_new_bet(self,name,statement,status='open'):
+        '''add a new bet'''
+        #load existing bets
+        with open("bet_log.json", "r") as read_file:
+            bet_log = json.load(read_file)
+        current_bet_id = bet_log['current_bet_id'] + 1
+        bet_key = 'bet_'+str(current_bet_id)
+
+        #make the new bet
+        new_bet = {
+            "bet_id": current_bet_id,
+            "bidder": name,
+            "status": status,
+            "statement": statement
+        }
+        bet_log[bet_key] = new_bet
+
+        #update current_bet_id
+        bet_log['current_bet_id'] = current_bet_id
+
+        #save updated bets
+        with open("bet_log.json","w") as write_file:
+            json.dump(bet_log,write_file,indent=4)
+        return bet_log
+
+    def check_author(self,ctx,bet_id: int):
+        '''checks if the person giving commands is in the bet'''
+        #load bets
+        with open("bet_log.json", "r") as read_file:
+            bet_log = json.load(read_file)
+        bet_key = 'bet_'+str(bet_id)
+
+        #check bidder and seller (if it exists)
+        try:
+            if 'seller' in bet_log[bet_key]:
+                return bet_log[bet_key]['bidder'] == ctx.author.id or bet_log[bet_key]['seller'] == ctx.author.id
+            else:
+                return bet_log[bet_key]['bidder'] == ctx.author.id
+        except KeyError as e:
+            raise Exception('Couldn\'t check author')
```

Comment:
as before, helpful to raise a more specific exception? and message it out to the channel?